lib/mixlib/install/backend/bintray.rb

```diff
@@ -0,0 +1,210 @@
+#
+# Author:: Patrick Wright (<patrick@chef.io>)
+# Copyright:: Copyright (c) 2016 Chef, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "mixlib/install/backend/base"
+
+#
+# Add method to Array class to support
+# searching for substrings that match against
+# the items in the Array
+#
+class Array
+  def fuzzy_include?(search_value, regex_format = "%s")
+    inject(false) do |is_found, array_value|
+      is_found || !!(search_value =~ /#{regex_format % array_value}/)
+    end
+  end
+end
+
+module Mixlib
+  class Install
+    class Backend
+      class Bintray < Base
+        ENDPOINT = "https://bintray.com".freeze
+
+        API_VERSION = "v1".freeze
+
+        BINTRAY_ORGANIZATION = "chef".freeze
+
+        BINTRAY_USERNAME = "mixlib-install@chef".freeze
+
+        BINTRAY_PASSWORD = "a83d3a2ffad50eb9a2230f281a2e19b70fe0db2d".freeze
+
+        DOWNLOAD_URL_ENDPOINT = "https://packages.chef.io".freeze
+
+        def endpoint
+          @endpoint ||= ENV.fetch("BINTRAY_ENDPOINT", ENDPOINT)
+        end
+
+        def org
+          @org ||= ENV.fetch("BINTRAY_ORGANIZATION", BINTRAY_ORGANIZATION)
+        end
+
+        # Create filtered list of artifacts
+        #
+        # @return [Array<ArtifactInfo>] list of artifacts for the configured
+        # channel, product name, and product version.
+        # @return [ArtifactInfo] arifact info for the configured
+        # channel, product name, product version and platform info
+        #
+        def info
+          artifacts = bintray_artifacts
+
+          if options.platform
+            artifacts.select! do |a|
+              a.platform == options.platform &&
+                a.platform_version == options.platform_version &&
+                a.architecture == options.architecture
+            end
+          end
+
+          artifacts.length == 1 ? artifacts.first : artifacts
+        end
+
+        private
+
+        #
+        # Create Bintray client
+        #
+        # @return [Mixlib::Install::Bintray::Client] Mixlib::Install::Bintray::Client instance
+        #
+        def client
+          @client ||= Mixlib::Install::Bintray::Client.new(
```

Comment:
Where is `Mixlib::Install::Bintray::Client` defined? I don't see it in this PR and I also don't see a require for it at the top of this file.